### ISSUE TITLE
Devcontainer: Matplotlib issue with collecting tests

### ISSUE THREAD

**Issue by callumforrester** [bluesky/bluesky#1693]
Using matplotlib with a tk backend, we see an error collecting the tests

The following people have reported, please paste your logs:
@AlexanderWells-diamond @noemifrisina @gilesknap

**Comment by AlexanderWells-diamond** [comment]
```
$ tox -e tests -- --collect-only
...

=========================================================================== 1650 tests collected in 6.13s ===========================================================================
Exception ignored in atexit callback: <bound method Gcf.destroy_all of <class 'matplotlib._pylab_helpers.Gcf'>>
Traceback (most recent call last):
  File "/venv/lib/python3.10/site-packages/matplotlib/_pylab_helpers.py", line 82, in destroy_all
    manager.destroy()
  File "/venv/lib/python3.10/site-packages/matplotlib/backends/_backend_tk.py", line 565, in destroy
    self.canvas._tkcanvas.after_cancel(self.canvas._idle_draw_id)
  File "/usr/local/lib/python3.10/tkinter/__init__.py", line 871, in after_cancel
    data = self.tk.call('after', 'info', id)
RuntimeError: main thread is not in main loop
```

This happens inside of the Devcontainer for me. I haven't tried outside of it.

**Comment by callumforrester** [comment]
Here's a workaround: `MPLBACKEND=agg tox -e tests`

**Comment by AlexanderWells-diamond** [comment]
Another possible fix is to add these libraries to the `Dockerfile`:

```
RUN apt-get update && apt-get install -y --no-install-recommends \
    graphviz ffmpeg libsm6 libxext6 libegl1 libqt5gui5 libxcb-cursor0 -y \
    && rm -rf /var/lib/apt/lists/*
```

Note that this will cause matplotlib to start running using Qt5, rather than Tk, so it doesn't really fix the Tk issue...

**Comment by callumforrester** [comment]
Could also combine this ticket with #1689 and get rid of the GUI during test collection altogether, unsure how @tacaswell will feel about that though.

**Comment by callumforrester** [comment]
Reviewers: @tacaswell @gilesknap

**Comment by tacaswell** [comment]
https://github.com/bluesky/bluesky/pull/1672 should fix the "not main thread" issue.

**Comment by callumforrester** [comment]
Fixed by #1672 and #1700
